corehq/apps/data_interfaces/models.py

```diff
@@ -203,6 +213,272 @@ def references_parent_case(self):
+        will not be copied.
+        """
+        allow_sms_surveys = convert_form_unique_id_function is not None
+        if not self.conditional_alert_can_be_copied(
+            allow_sms_surveys=allow_sms_surveys,
+            allow_custom_references=allow_custom_references,
+        ):
+            return None
+
+        with transaction.atomic():
+            new_rule = AutomaticUpdateRule.objects.create(
+                domain=to_domain,
+                name=self.name,
+                case_type=self.case_type,
+                active=self.active,
+                last_run=self.last_run,
+                filter_on_server_modified=self.filter_on_server_modified,
+                server_modified_boundary=self.server_modified_boundary,
+                workflow=self.workflow,
+            )
+
+            for criterion in self.memoized_criteria:
+                definition = criterion.definition
+                if isinstance(definition, MatchPropertyDefinition):
+                    new_rule.add_criteria(
+                        MatchPropertyDefinition,
+                        property_name=definition.property_name,
+                        property_value=definition.property_value,
+                        match_type=definition.match_type,
+                    )
+                elif isinstance(definition, CustomMatchDefinition):
+                    new_rule.add_criteria(
+                        CustomMatchDefinition,
+                        name=definition.name,
+                    )
+                else:
+                    raise TypeError(
+                        "Unexpected criteria definition. Did conditional_alert_can_be_copied() get called?"
+                    )
+
+            action_definition = self.get_messaging_rule_action_definition()
+            schedule = action_definition.schedule
+
+            new_schedule = self.copy_schedule(schedule, to_domain,
+                convert_form_unique_id_function=convert_form_unique_id_function)
+
+            new_rule.add_action(
+                CreateScheduleInstanceActionDefinition,
+                alert_schedule_id=new_schedule.schedule_id if isinstance(new_schedule, AlertSchedule) else None,
+                timed_schedule_id=new_schedule.schedule_id if isinstance(new_schedule, TimedSchedule) else None,
+                recipients=deepcopy(action_definition.recipients),
+                reset_case_property_name=action_definition.reset_case_property_name,
+                start_date_case_property=action_definition.start_date_case_property,
+                specific_start_date=action_definition.specific_start_date,
+                scheduler_module_info=deepcopy(action_definition.scheduler_module_info),
+            )
+
+        return new_rule
+
+    def fix_sms_survey_reference(self, copied_content, original_content, convert_form_unique_id_function):
+        copied_content.form_unique_id = convert_form_unique_id_function(original_content.form_unique_id)
```

Comment:
Should this check for None? It looks like `copy_conditional_alert` expects `convert_form_unique_id_function` to be None sometimes (based on the creation of `allow_sms_surveys`) and then passes it on to `copy_schedule` and then here without checking.

Reply:
we don't need to - allow_sms_surveys will be False if convert_form_unique_id_function is None so we'll never attempt copying sms surveys unless the function is not None

Reply:
okay, cool